app/workers/send_additional_reference_chase_email_to_both_parties_worker.rb

```diff
@@ -22,11 +22,19 @@ def applications_that_need_chasing
       .feedback_requested
       .where(['requested_at < ?', time_limit])
 
-    application_form_ids =
-      outstanding_reference_requests
-      .where(application_form_id: applications_waiting_on_reference_feedback.pluck(:id))
-      .distinct(:application_form_id)
-      .pluck(:application_form_id)
+    if FeatureFlag.active?(:decoupled_references)
+      application_form_ids =
+        outstanding_reference_requests
+        .where(application_form_id: unsubmitted_application_forms.pluck(:id))
+        .distinct(:application_form_id)
+        .pluck(:application_form_id)
```

Comment:
Same as above. Does it make sense?

Reply:
Agreed, this seems like a good opportunity to move that logic into something like a `GetReferencesToChaseAgain` (or similar) query object. There's probably some further refactoring to be done there, but at least this way it's clear that two similar pieces of logic are involved and they're cleanly abstracted away.

Another issue (outside the scope of this PR) is that we model each chase differently. The first is recorded as a chase on the reference, the second is a chase on the entire application form. Can't recall why we did it this way but worth revisiting.